### PR TITLE
Lock the reports list in plank.

### PR DESCRIPTION
**Pull request by spxtr**
This fixes a race when the sync functions run in parallel.

cc @stevekuznetsov

### CODE REVIEW

File: prow/plank/controller.go [hunk 1]
```diff
@@ -276,7 +282,9 @@ func (c *Controller) syncJenkinsJob(pj kube.ProwJob) error {
 			pj.Status.JenkinsEnqueued = true
 			pj.Status.Description = "Jenkins job triggered."
 		}
+		c.reportsLock.Lock()
```
Comment (fejta) [inline prow/plank/controller.go]: Would be worthwhile encapsulating this inside an appendReport() type method?

**Comment by stevekuznetsov** [comment]
Sorry I didn't catch this either in the first PR... I can take a stab at channel-izing it if you woud like.

/lgtm

File: prow/plank/controller.go (hunk 1 above)
Comment (stevekuznetsov) [inline prow/plank/controller.go]: Could shove this into an append method to reduce the lock/unlock copy-paste

Comment (spxtr) [inline prow/plank/controller.go]: Done.

Comment (spxtr) [inline prow/plank/controller.go]: Done.

**Comment by spxtr** [comment]
> Sorry I didn't catch this either in the first PR... I can take a stab at channel-izing it if you would like.

No worries, and I'd be happy if you did :)

**Comment by stevekuznetsov** [comment]
Pull at #3816

**Pull request merged**